### PR TITLE
Add Datadog registry entries in flare

### PR DESCRIPTION
**Pull request by iglendd**
### What does this PR do?
Adds datadog.reg file to the Flare's package on Windows. The file content is exported SOFTWARE\Datadog registry key.

* PR added a code which invokes built in Windows command`reg export HKLM\Software\Datadog  <file path\datadog.reg> /y`

### Motivation
Troubleshooting assistance.

### Additional Notes
No unit or kitchen tests had been provided due to ...
* ... very simple implementation
* ... not much value from decoupling code from concrete registry path
* ... no similar flare artifacts had been tested in unit tests
* ... flare is not tested in kitchen tests

### Possible Drawbacks / Trade-offs
There are two ways to implement this functionality with its own pros and cons.
#### Using Registry API calls
One would use [Registry API ](https://pkg.go.dev/golang.org/x/sys/windows/registry) to recursively enumerate and  dump Keys and Values. While the solution guarantees high performance (perhaps handful of microseconds) interpreting  Registry value to be saved as text requires relatively big effort to handle [all nearly a dozen types of Registry values](https://docs.microsoft.com/en-us/windows/win32/sysinfo/registry-value-types). Consider a relatively trivial example of various registry value types
```
[HKEY_LOCAL_MACHINE\Software\Datadog\Len-Test]
"str"="asdfasdf"
"bin"=hex:ad,fd,fa,a0
"dword"=dword:00000000
"multi-str"=hex(7):61,00,73,00,64,00,66,00,61,00,73,00,64,00,66,00,00,00,61,00,73,\
  00,64,00,66,00,61,00,73,00,64,00,66,00,66,00,66,00,66,00,66,00,66,00,00,00,\
  73,00,73,00,73,00,73,00,00,00,73,00,73,00,73,00,00,00,00,00
"expand-str"=hex(2):25,00,61,00,70,00,70,00,64,00,61,00,74,00,61,00,25,00,00,00

```
#### Spawning Windows built-in reg.exe command
Starting [reg.exe](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/reg) process had a relatively large overhead of small executable start but fast otherwise (perhaps in order of few milliseconds). However, zero coding is needed to capture Datadog all registry key and values in the proper format directly in the file. Because the file is saved in the protected Flare's directory (where only datadog user can access it) solution is secure.

### Describe how to test/QA your changes
Run flare command on Windows. Observe the presence of datadog.reg file in the flare package.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a release note.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by albertvaka — APPROVED** [review]
LGTM

File: pkg/flare/archive_win.go [hunk 1]
```diff
@@ -259,3 +259,61 @@ func zipServiceStatus(tempDir, hostname string) error {
...
+//            00,64,00,66,00,61,00,73,00,64,00,66,00,66,00,66,00,66,00,66,00,66,00,00,00,\
+//            73,00,73,00,73,00,73,00,00,00,73,00,73,00,73,00,00,00,00,00
+//          "expand-str"=hex(2):25,00,61,00,70,00,70,00,64,00,61,00,74,00,61,00,25,00,00,00
+//
+// 2. Spawning reg.exe process (https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/reg)
+//    It is builtin command and will do the heavy lifting of the recursion and value dumping but it
+//    creating process is relatively slow and generate a file which need to be scrubbed. From
+//    security standpoint it is still secure since generated file can be read only by an administrator
+//    or a ddagentuser.
+//
```
[the remark this answers is not in the document]
Comment (iglendd) [reply pkg/flare/archive_win.go:284]: @albertvaka I do not mind add this to PR description but when someone is looking on the code later they will not see that longer explanation in context. Of course I do not have a feel yet what is a general style of addressing this kind of concern and I would rely on your judgement. Please confirm it.

[the remark this answers is not in the document]
Comment (djmitche) [reply pkg/flare/archive_win.go:284]: You could achieve both by briefly summarizing the decision in the comment and either referencing the PR "for more details", or including this longer text in the commit message.  PRs and commit messages represent changes to the code, so they're a good place to store information about the changes.

[the remark this answers is not in the document]
Comment (iglendd) [reply pkg/flare/archive_win.go:284]: Sounds good, thank you.

File: releasenotes/notes/add-registry-to-flare-bb2b806f374633e8.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Windows: Add Datadog registry to flare..
```
Comment (alai97) [inline releasenotes/notes/add-registry-to-flare-bb2b806f374633e8.yaml:11]:
```suggestion
    Windows: Add Datadog registry to Flare.
```

**Review by alai97 — APPROVED** [review]
Looks good, one copy nit!

File: releasenotes/notes/add-registry-to-flare-bb2b806f374633e8.yaml (hunk 2 above)
Comment (iglendd) [reply releasenotes/notes/add-registry-to-flare-bb2b806f374633e8.yaml:11]: Thank you, on it

Comment (iglendd) [reply releasenotes/notes/add-registry-to-flare-bb2b806f374633e8.yaml]: Fix it, but just realized that the commit comment contains `flare` in low-case. Hopefully will be more vigilant next time.

**Review by djmitche — APPROVED** [review]
The implementation looks good to me, and I'm glad to see it's scrubbed.

File: pkg/flare/archive_win.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (djmitche) [reply pkg/flare/archive_win.go]: First comment on https://lwn.net/Articles/888413/ is relevant to this question, btw -- look at the size of those commit comments!

**Pull request merged**